ddtrace/contrib/algoliasearch/patch.py

```diff
@@ -125,7 +125,7 @@ def _patched_search(func, instance, wrapt_args, wrapt_kwargs):
         span.set_tag_str(SPAN_KIND, SpanKind.CLIENT)
 
         span.set_tag(SPAN_MEASURED_KEY)
-        if span.context.sampling_priority is None or span.context.sampling_priority <= 0:
+        if span.context.sampling_priority is not None and span.context.sampling_priority <= 0:
```

Comment:
Why do we even need to check the sampling priority in an integration?

Reply:
I think the idea was to avoid unnecessary and possibly expensive operations (mostly getting and setting tags) on a span if we knew it was going to be dropped. This brings up some additional questions as to if we should do this anymore now that we know single span sampling could choose to keep etc. However, I think that's out of the scope of this PR, so I'd prefer just trying to maintain behavior as well as possible. 

Although with lazy sampling the extra tag setting code will run more often, since if sampling hasn't run yet, or there isn't an upstream decision, we'll add the tags, so there may be a small performance impact.

I think a follow up PR could remove this behavior all together, but that would come a with a worse performance impact on average, since we'd always set the tags, even if there was an upstream drop.

I guess another thing to think about is if  single span sampling is used to keep these spans, after they've decided to be dropped, and didn't have tags added that could be confusing. I'll add this as a discussion topic for sync to discuss what's best, but don't think this should block this PR.